src/drive/mobile/lib/intents.js

```diff
@@ -111,3 +123,48 @@ export const intentHandler = store => async ({
     }
   }
 }
+
+const utiToMime = {
+  'com.adobe.pdf': 'application/pdf',
+  'com.adobe.postscript': 'application/postscript',
+  'com.compuserve.gif': 'image/gif',
+  'com.microsoft.bmp': 'image/bmp',
+  'com.microsoft.word.doc': 'application/msword',
+  'com.microsoft.excel.xls': 'application/vnd.ms-excel',
+  'com.microsoft.powerpoint.ppt': 'application/mspowerpoint'
+  // see https://developer.apple.com/library/content/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html#//apple_ref/doc/uid/TP40009259-SW1
+}
+
+const getMimeTypeFromUTI = utis => {
+  const knowUTIs = Object.keys(utiToMime)
+  const firstKnownUTI = knowUTIs.find(uti => utis.indexOf(uti) >= 0)
+
+  return firstKnownUTI ? utiToMime[firstKnownUTI] : ''
+}
+
+export const intentHandlerIOS = store => async intent => {
+  const files = await Promise.all(
+    intent.items.map(async item => {
+      if (item.uti === 'public.image') {
+        const blob = base64toBlob(item.data, item.type)
+        blob.name = item.name
+        return blob
+      } else {
+        const dirEntry = await getEntry(item.data)
+        if (!dirEntry.type) dirEntry.type = getMimeTypeFromUTI(item.utis)
```

Comment:
For files that are not images, we receive a link to the file, and *sometimes* a mime type, but mostly not. But we do get [UTIs](https://developer.apple.com/library/content/documentation/Miscellaneous/Reference/UTIRef/Articles/System-DeclaredUniformTypeIdentifiers.html#//apple_ref/doc/uid/TP40009259-SW1), so i've mapped the main ones to mime types.